crates/transaction-emitter-lib/src/emitter/account_minter.rs

```diff
@@ -277,14 +280,28 @@ impl<'t> AccountMinter<'t> {
         amount: u64,
     ) -> Result<()> {
         info!("Minting new coins to root");
+        for i in 0..3 {
+            self.source_account.set_sequence_number(
+                txn_executor
+                    .query_sequence_number(self.source_account.address())
+                    .await?,
+            );
 
-        let txn = self
-            .source_account
-            .sign_with_transaction_builder(self.txn_factory.payload(
-                aptos_stdlib::aptos_coin_mint(self.source_account.address(), amount),
-            ));
-        txn_executor.execute_transactions(&[txn]).await?;
-        Ok(())
+            let txn = self
+                .source_account
+                .sign_with_transaction_builder(self.txn_factory.payload(
+                    aptos_stdlib::aptos_coin_mint(self.source_account.address(), amount),
+                ));
+            match txn_executor.execute_transactions(&[txn]).await {
+                Ok(_) => return Ok(()),
+                Err(e) => {
+                    error!("Couldn't mint to root, {:?}, try {}, retrying", e, i);
+                    // random sleep to coordinate with other instances
+                    std::thread::sleep(Duration::from_secs(rand::thread_rng().gen_range(0, 10)));
```

Comment:
If it's async, you should use Tokio sleep vs thread sleep.  Otherwise, the whole thread's tasks will sleep for that long.  10 seconds is a long time.  If you're creating 10k addresses and this happens every time, you could wait 100k seconds, which is greater than 3 hours

Reply:
these two positions we do only once per emitter run
good point on async sleep 👍